Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabaseTransaction.cpp

```diff
@@ -126,14 +126,17 @@ void UniqueIDBDatabaseTransaction::commit(uint64_t pendingRequestCount)
         return;
 
     std::optional<IDBError> errorInPendingRequests;
-    while (pendingRequestCount--) {
+    while (pendingRequestCount-- && !m_requestResults.isEmpty()) {
         auto error = m_requestResults.takeLast();
```

Comment:
we append to `m_requestResults` asynchronously (after the IPC receipt) in this class. However, we remove synchronously (after IPC receipt) in this function. Is it possible the root cause of this is simply a race?

Note that we need your hardening either way since the `pendingRequestCount` may come from IPC.

Reply:
Fixed in updated patch.